plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -517,6 +521,36 @@ instance KnownBuiltinTypeIn DefaultUni term Integer => ReadKnownIn DefaultUni te
                  ]
     {-# INLINE readKnown #-}
 
+outOfBoundsErr :: Pretty a => a -> Vector.Vector term -> Text
+outOfBoundsErr x branches = fold
+    [ "'case "
+    , display x
+    , "' is out of bounds for the given number of branches: "
+    , display $ Vector.length branches
+    ]
+
+instance AnnotateCaseBuiltin DefaultUni where
+    annotateCaseBuiltin (SomeTypeIn uni) branches = case uni of
+        DefaultUniBool    -> Right $ map (, []) branches
+        DefaultUniInteger -> Right $ map (, []) branches
+        _                 -> Left $ display uni <> " isn't supported in 'case'"
+
+instance UniOf term ~ DefaultUni => CaseBuiltin term DefaultUni where
+    caseBuiltin (Some (ValueOf uni x)) branches = case uni of
```

Comment:
Don't you want to `INLINE` this?

Reply:
Doesn't really matter, because there's no place where it'd be useful to inline it, but I can annotate it with `INLINE` just in case it becomes useful in future. Thanks!